### ISSUE TITLE
[BUG] DataFrame.iloc crashes for string column with None

### ISSUE THREAD

**Issue by karthikeyann** [rapidsai/cudf#2000]
**Description of the bug**
When list is passed to DataFrame.iloc for dataframe with string column with None, it crashes.

**Steps/Code to reproduce bug**
```
import numpy as np
from pandas import DataFrame
import cudf

pdf = DataFrame({'A': [None, None, 'foo', 'bar', 'foo', 'baz', 'bar', 'qux']})
df = cudf.DataFrame.from_pandas(pdf)
print(df.iloc[range(8)])
print(df.iloc[list(range(8))])
```

**Crash Error:**
terminate called after throwing an instance of 'thrust::system::system_error'
  what():  parallel_for failed: invalid device pointer
Aborted

**Expected behavior**
Result of both `print(df.iloc[range(8)])` and `print(df.iloc[list(range(8))])` should be equal.
```
     A
0  None
1  None
2   foo
3   bar
4   foo
5   baz
6   bar
7   qux
```

**Environment details:**
 Bare-metal, cudf from source

**Comment by karthikeyann** [comment]
does PR https://github.com/rapidsai/cudf/pull/2073 fix this issue?

**Comment by harrism** [comment]
@shwina does `iloc` call cudf::gather?

**Comment by kkraus14** [comment]
@harrism under the hood it does. This should be resolved with #2073.

**Comment by kkraus14** [comment]
Confirmed fixed with #2073.
